CHANGELOG.md

```diff
@@ -2,6 +2,73 @@
+- Semantic attributes `code.namespace` and `code.function` captured on JAX-RS internal spans
+  ([#2805](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2805))
+- `SERVER` span names improved for Spring MVC, Grails, Wicket, and Struts
+  ([#2814](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2814))
+- `SERVER` span names improved for Servlet filters
+  ([#2887](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2887))
+- `SERVER` span names improved for Resteasy
+  ([#2900](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2900))
+- `SERVER` span names improved for Jersey and CXF
+  ([#2919](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2919))
+- JAX-RS `@ApplicationPath` annotation captured as part of `SERVER` span name
+  ([#2824](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2824))
+- RequestDispatcher `forward()` and `include()` internal spans removed
+  ([#2816](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2816))
+- Context propagated to reactor-netty callbacks
+  ([#2850](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2850))
+- AWS lambda flush timeout raised to 10 seconds
+  ([#2855](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2855))
+
+### 🛠️ Bug fixes
+
+- RxJava2 NoSuchFieldError
+  ([#2836](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2836))
+- Duplicate http client tracing headers
+  ([#2842](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2842))
+- Netty 4.1 listeners could not be removed by application
+  ([#2851](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2851))
+- NPE caused in gRPC ProtoReflectionService
+  ([#2876](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2876))
+- Context leak when using Ratpack
+  ([#2910](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2910))
+- Context leak when uising Jetty
+  ([#2920](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2920))
+
+### 🧰 Tooling
+
+- Documented how to write InstrumentationModule line by line
+  ([#2793](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2793))
+- New instrumenter API used in JMS instrumentation
+  ([#2803](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2803))
+- Instrumenter API improvements
+  ([#2860](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2860))
+- Muzzle checks whether used fields are actually declared somewhere
+  ([#2870](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2870))
+- Extracted javaagent-extension-api from tooling & spi
+  ([#2879](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2879))
```

Comment:
Should we provide some details about the breaking changes here?

```suggestion
- Extracted javaagent-extension-api from tooling & spi
  ([#2879](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2879))
  - You no longer have to depend on the `javaagent-tooling` module to implement custom instrumentations: a new `javaagent-extension-api` module was introduced, containing all the necessary instrumentation classes and interfaces;
  - `InstrumentationModule` and `TypeInstrumentation` were moved to the `io.opentelemetry.javaagent.extension.instrumentation` package;
  - `AgentElementMatchers`, `ClassLoaderMatcher` and `NameMatchers` were moved to the `io.opentelemetry.javaagent.extension.matcher` package;
  - A new SPI `AgentExtension` was introduced: it will replace `ByteBuddyAgentCustomizer` in the next release.
```